connector/src/main/java/org/geysermc/connector/network/session/UpstreamSession.java

```diff
@@ -51,11 +56,24 @@ public void sendPacketImmediately(@NonNull BedrockPacket packet) {
         if (isClosed())
             return;
 
+        packet.setSenderId(clientId);
         session.sendPacketImmediately(packet);
     }
 
     public void disconnect(String reason) {
-        session.disconnect(reason);
+        if (isClosed()) {
```

Comment:
Not sure why this needs to be reimplemented - since it doesn't pass through GeyserSession's `disconnect` function it doesn't properly null all the values of the class.

Reply:
Because it needs to add that senderId, which isn't done by the external implementation.

Reply:
Is it possible to move it to that external implementation?

Reply:
There's also no easy way to get the clientId in that function, because BedrockServerSession isn't aware of different disconnect message for different subclients at this point. I'll PR that for an overload with an extra argument.

Reply:
Oh and if we don't set the right clientId, the MainClient is disconnected when the SubClient leaves the game.

Reply:
This will be resolved when done in Protocol instead.